bin/ecompress

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+source "${PORTAGE_BIN_PATH}"/isolated-functions.sh || exit 1
+
+if [[ -z $1 ]] ; then
+	__helpers_die "${0##*/}: at least one argument needed"
+	exit 1
+fi
+
+if ! ___eapi_has_prefix_variables; then
+	ED=${D} EPREFIX=
+fi
+
+while [[ $# -gt 0 ]] ; do
+	case $1 in
+	--ignore)
+		shift
+
+		skip_dirs=()
+		for skip; do
+			if [[ -d ${ED%/}/${skip#/} ]]; then
+				skip_dirs+=( "${ED%/}/${skip#/}" )
+			else
+				rm -f "${ED%/}/${skip#/}.ecompress" || die
+			fi
+		done
+
+		if [[ ${skip_dirs[@]} ]]; then
+			find "${skip_dirs[@]}" -name '*.ecompress' -delete || die
```

Comment:
PMS doesn't say anything about symlinks in the context of compression, so from that point of view there should be no problems.

The new code will indeed behave differently, but I wonder how often symlinks between directories occur in practice? If they do, how many ebuilds use the symlinked directory instead of the physical directory as an argument to ``docompress``? @mgorny, could you scan the tree to find that out? (If the answer turns out to be zero, we need not worry about it.)

The one case that is very common are symlinks between man pages. For those, both the name of the symlink and the link target must be updated (namely, the compress extension added). Does the new code handle that correctly?

It may also be worthwhile to look at the discussion in https://bugs.gentoo.org/260118#c19 (up to comment 24) again.

Reply:
This only affects symlinks as direct arguments to `docompress -x`, i.e. while I haven't checked all the ebuilds individually, I think they are none relying on this (why would you?). After all, most of them just pass it on some directory in `/usr/share/doc/${PF}`

Reply:
(and yes, the code handles symlinks between manpages, that's its purpose)

Reply:
(I mean the symlink fixing code)